src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Check.java

```diff
@@ -4908,6 +4909,92 @@ private boolean patternDominated(JCPattern existingPattern, JCPattern currentPat
             return false;
         }
 
+    void checkDerivedInstanceBlockStructure(JCDerivedInstance instance) {
+        new TreeScanner() {
```

Comment:
I note that here we don't care about leaving locals declared inside a block in the `seenVariables` after the block ends (e.g. I see no visitBlock). I think that's probably ok, given that every variable has a fresh symbol, so checking against the `seenVariables` set will fail anyway for two variables that have same name, but different scopes.